File: src/test/java/com/github/korosuke613/bwdm/informationStore/IfElseExprSyntaxTreeTest.java
package com.github.korosuke613.bwdm.informationStore;

import com.fujitsu.vdmj.ast.definitions.ASTDefinition;
import com.fujitsu.vdmj.ast.definitions.ASTDefinitionList;
import com.fujitsu.vdmj.lex.Dialect;
import com.fujitsu.vdmj.lex.LexException;
import com.fujitsu.vdmj.lex.LexTokenReader;
import com.fujitsu.vdmj.mapper.ClassMapper;
import com.fujitsu.vdmj.syntax.DefinitionReader;
import com.fujitsu.vdmj.syntax.ParserException;
import com.fujitsu.vdmj.tc.definitions.TCDefinition;
import com.fujitsu.vdmj.tc.definitions.TCExplicitFunctionDefinition;
import com.fujitsu.vdmj.tc.expressions.TCExpression;
import java.io.File;
import java.io.IOException;
import org.junit.jupiter.api.*;

class IfElseExprSyntaxTreeTest {

  static private IfElseExprSyntaxTree ifElseExprSyntaxTree;

  @BeforeAll
  static void initAll() throws LexException, ParserException {

    LexTokenReader lexer = new LexTokenReader(
        new File("./vdm_files/Arg2_Japanese.vdmpp"), Dialect.VDM_PP);
    DefinitionReader parser = new DefinitionReader(lexer);
    ASTDefinitionList astDefinitions = parser.readDefinitions();

    astDefinitions.forEach((ASTDefinition astDefinition) -> {
      if (astDefinition.kind().equals("explicit function")) {
        TCExplicitFunctionDefinition tcFunctionDefinition = null;
        try {
          tcFunctionDefinition = ClassMapper.getInstance(TCDefinition.MAPPINGS)
                                     .init()
                                     .convert(astDefinition);
        } catch (Exception e) {
          e.printStackTrace();
        }
        assert tcFunctionDefinition != null;
        TCExpression tcExpression = tcFunctionDefinition.body;
        String ifExpressionBody = tcExpression.toString();

        ifElseExprSyntaxTree = null;
        ifElseExprSyntaxTree = new IfElseExprSyntaxTree(ifExpressionBody);
      }
    });
  }

  @BeforeEach
  void init() {}

  @Test
  void ほげ() {
    assert IfElseExprSyntaxTree.class.isInstance(ifElseExprSyntaxTree);
  }

  @AfterEach
  void tearDown() {}

  @AfterAll
  static void tearDownAll() {}
}

File: src/test/kotlin/com/github/korosuke613/bwdm/informationStore/IfElseExprSyntaxTreeTest.kt
package com.github.korosuke613.bwdm.informationStore

import com.fujitsu.vdmj.ast.definitions.ASTDefinition
import com.fujitsu.vdmj.lex.Dialect
import com.fujitsu.vdmj.lex.LexException
import com.fujitsu.vdmj.lex.LexTokenReader
import com.fujitsu.vdmj.mapper.ClassMapper
import com.fujitsu.vdmj.syntax.DefinitionReader
import com.fujitsu.vdmj.syntax.ParserException
import com.fujitsu.vdmj.tc.definitions.TCDefinition
import com.fujitsu.vdmj.tc.definitions.TCExplicitFunctionDefinition
import org.junit.jupiter.api.*
import java.io.File
import java.util.function.Consumer

internal class IfElseExprSyntaxTreeTest {
    @BeforeEach
    fun init() {
    }

    @Test
    fun ほげ() {
        assert(IfElseExprSyntaxTree::class.java.isInstance(ifElseExprSyntaxTree))
    }

    @AfterEach
    fun tearDown() {
    }

    companion object {
        private var ifElseExprSyntaxTree: IfElseExprSyntaxTree? = null
        @BeforeAll
        @Throws(LexException::class, ParserException::class)
        fun initAll() {
            val lexer = LexTokenReader(
                    File("./vdm_files/Arg2_Japanese.vdmpp"), Dialect.VDM_PP)
            val parser = DefinitionReader(lexer)
            val astDefinitions = parser.readDefinitions()
            astDefinitions.forEach(Consumer { astDefinition: ASTDefinition ->
                if (astDefinition.kind() == "explicit function") {
                    var tcFunctionDefinition: TCExplicitFunctionDefinition? = null
                    try {
                        tcFunctionDefinition = ClassMapper.getInstance(TCDefinition.MAPPINGS)
                                .init()
                                .convert<TCExplicitFunctionDefinition>(astDefinition)
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                    assert(tcFunctionDefinition != null)
                    val tcExpression = tcFunctionDefinition!!.body
                    val ifExpressionBody = tcExpression.toString()
                    ifElseExprSyntaxTree = null
                    ifElseExprSyntaxTree = IfElseExprSyntaxTree(ifExpressionBody)
                }
            })
        }

        @AfterAll
        fun tearDownAll() {
        }
    }
}